src/libraries/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -164,9 +164,10 @@ public void Insert(int index, string? s)
         public void Append(char c)
         {
             int pos = _pos;
-            if ((uint)pos < (uint)_chars.Length)
+            Span<char> chars = _chars;
```

Comment:
I would expect that the JIT CSE optimizations will do this for you when it is worth it. Do you see this making a difference?

Reply:
Also, the changes in this method aren't about allocations.  If part of the goal of this PR is throughput, there should be a perf test/results shared as part of the PR.